solr/solrj/src/java/org/apache/solr/common/cloud/PerReplicaStates.java

```diff
@@ -130,6 +140,26 @@ public State get(String replica) {
     return states.get(replica);
   }
 
+  /**
+   * A new child node got created. construct a new Object with this
+   *
+   * @param newState the newly inserted state
+   * @param czxid create tx id of the child node
+   */
+  public PerReplicaStates insert(State newState, long czxid) {
+    State existing = states.get(newState.replica);
+    if (existing == null || existing.version < newState.version) {
+      LinkedHashMap<String, State> copy = new LinkedHashMap<>(states.size());
+      states.forEachEntry(copy::put);
+      copy.put(newState.replica, newState);
+      return new PerReplicaStates(
+          path,
+          new WrappedSimpleMap<>(copy),
+          czxid); // child node's czxid is same as state.json pzxid
+    }
```

Comment:
Probably want to add a case 
```
} else if (newState.asString.equals(existing.asString)) {
      return this;
    }
```
here to avoid unnecessary fetch. With the current design, the existing state can be the same as the new state